src/backend/distributed/commands/table.c

```diff
@@ -418,11 +418,15 @@ PlanAlterTableStmt(AlterTableStmt *alterTableStatement, const char *alterTableCo
 															   command);
 	}
 
+	if (executeSequentially)
```

Comment:
Deferring this decision leads to weird edge cases on adaptive executor because there is an important reference to `SEQUENTIAL_CONNECTION` on `UseConnectionPerPlacement()`.

I think this is easier to follow and makes things more robust.